MC-21773: [MFTF] Failing Test: EndToEndB2CGuestUserMysqlTest

abothu opened the pull request:
"This E2E test is not part of Inventory Module but it is part of CE,EE,B2B module and uses the extension to test Inventory Module as well. We recently made changes to original test "EndToEndB2CGuestUserTest" as part of Elastic Search support. But at the same time we clone original test EndToEndB2CGuestUserTest and created a new one EndToEndB2CGuestUserMysqlTest to support the default Mysql Search Engine as well. While making changes we forgot to make changes of adding EndToEndB2CGuestUserMysqlTest on MSI side as well which causes our nightly jobs to fail as it runs on MSI extensions. I made necessary changes to fix this test"

<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. ...
2. ...

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

abothu committed:
MC-21773: [MFTF] Failing Test: EndToEndB2CGuestUserMysqlTest
"This E2E test is not part of Inventory Module but it is part of CE,EE,B2B module and uses the extension to test Inventory Module as well. We recently made changes to original test "EndToEndB2CGuestUserTest" as part of Elastic Search support. But at the same time we clone original test EndToEndB2CGuestUserTest and created a new one EndToEndB2CGuestUserMysqlTest to support the default Mysql Search Engine as well. While making changes we forgot to make changes of adding EndToEndB2CGuestUserMysqlTest on MSI side as well which causes our nightly jobs to fail as it runs on MSI extensions. I made necessary changes to fix this test"

sidolov commented on app/code/Magento/Catalog/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml:
```diff
@@ -17,6 +17,198 @@
             <description value="User browses catalog, searches for product, adds product to cart, adds product to wishlist, compares products, uses coupon code and checks out."/>
             <severity value="CRITICAL"/>
             <testCaseId value="MAGETWO-87435"/>
+            <group value="SearchEngineElasticsearch"/>
```
Please, revert changes, files in the Catalog module are part of Magento core, not MSI project

abothu replied:
Reverted changes

sidolov commented on app/code/Magento/CatalogSearch/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml:
```diff
@@ -65,6 +65,72 @@
         <grabAttributeFrom selector="{{StorefrontCategoryProductSection.ProductImageByName($$createSimpleProduct2.name$$)}}" userInput="src" stepKey="searchGrabSimpleProduct2ImageSrc" after="searchAssertFilterCategorySimpleProduct2"/>
         <assertNotRegExp expected="'/placeholder\/small_image\.jpg/'" actual="$searchGrabSimpleProduct2ImageSrc" stepKey="searchAssertSimpleProduct2ImageNotDefault" after="searchGrabSimpleProduct2ImageSrc"/>
 
+        <!-- Quick Search with non-existent product name -->
```
Please, revert changes, files in the CatalogSearch module are part of Magento core, not MSI project

abothu replied:
Reverted changes

sidolov commented on app/code/Magento/Checkout/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml:
```diff
@@ -184,6 +184,194 @@
             <argument name="productVar" value="$$createSimpleProduct2$$"/>
         </actionGroup>
 
+        <comment userInput="Place order with check money order payment" stepKey="commentPlaceOrderWithCheckMoneyOrderPayment" after="guestCheckoutCheckSimpleProduct2InCartItems" />
```
Please, revert changes, files in the Checkout module are part of Magento core, not MSI project

abothu replied:
Reverted changes

sidolov commented on app/code/Magento/ConfigurableProduct/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml:
```diff
@@ -9,6 +9,215 @@
 <tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
     <test name="EndToEndB2CGuestUserTest">
+    <before>
```
Please, revert changes, files in the ConfigurableProduct module are part of Magento core, not MSI project

abothu replied:
Reverted changes

sidolov commented on app/code/Magento/SalesRule/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml:
```diff
@@ -38,6 +38,45 @@
             <argument name="total" value="447.00"/>
         </actionGroup>
 
+        <actionGroup ref="StorefrontCancelCouponActionGroup" stepKey="couponCancelCoupon" after="couponCheckCartWithDiscount"/>
```
Please, revert changes, files in the SalesRule module are part of Magento core, not MSI project

abothu replied:
Reverted changes

sidolov commented on dev/tests/acceptance/tests/functional/Magento/FunctionalTest/ConfigurableProductCatalogSearch/Test/EndToEndB2CGuestUserTest.xml:
```diff
@@ -27,4 +27,23 @@
         <grabAttributeFrom selector="{{StorefrontProductInfoMainSection.productImage}}" userInput="src" stepKey="searchGrabConfigProductPageImageSrc" after="searchAssertConfigProductPage"/>
         <assertNotRegExp expected="'/placeholder\/image\.jpg/'" actual="$searchGrabConfigProductPageImageSrc" stepKey="searchAssertConfigProductPageImageNotDefault" after="searchGrabConfigProductPageImageSrc"/>
     </test>
+    <test name="EndToEndB2CGuestUserMysqlTest">
```
Please, revert changes, files in the ConfigurableProductCatalogSearch module are part of Magento core, not MSI project

abothu replied:
Reverted changes

abothu committed:
MC-21773: [MFTF] Failing Test: EndToEndB2CGuestUserMysqlTest
"Reverted above files based on the review comments from https://github.com/magento/inventory/pull/2629"

abothu commented:
Closing this PR and created a new one.
